BatteryReporter.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace BatteryCharging
+{
+    public interface IReporter
+    {
+        void DisplayBatteryStatus(bool IsBatteryFaulty, List<Tuple<string, string>> batteryReports);
+    }
+    public class BatteryReporter : IReporter
+    {
+        public void DisplayBatteryStatus(bool IsBatteryFaulty, List<Tuple<string, string>> batteryReports)
+        {
+            foreach (var item in batteryReports)
+            {
+                Console.WriteLine("{0}\n{1}", item.Item1, item.Item2.ToString());
+            }
+            if (IsBatteryFaulty)
+            {
```

Comment:
Redundant Code --- Console.WriteLine 
If class responsibility is to report , then what is the need of checking IsBatteryFaulty

Reply:
Redundant code and condition check is removed